### PR TITLE
git-combine: attempt to clean up stale git lockfiles from crashed processes

### PR DESCRIPTION
**Pull request by ggilmore**
If git-combined crashes/restarts in the middle of a git operation, it can leave behind stale lockfiles (like index.lock, `refs/remotes/origin/main.lock`, etc. that can prevent git from running properly when the pod restarts:

```bash
/data/repos/gigarepo/.git # git gc
^Z[1]+  Stopped                    git gc
/data/repos/gigarepo/.git # kill %1
/data/repos/gigarepo/.git # git gc
fatal: gc is already running on machine 'git-combine-0' pid 10968 (use --force if not)
```

This PR attempts to remove all such "stale" git lockfiles when the daemon first starts, so that the pod avoids an endless crash loop. 

## Test plan

This is a PR for a utility command outside of the sourcegraph/sourcegraph codebase.

### CODE REVIEW

**Comment by ggilmore** [comment]
cc @keegancsmith. I haven't tested this yet, but I think this is in a good enough state for review.

File: internal/cmd/git-combine/git-combine.go [hunk 1]
```diff
@@ -440,6 +448,47 @@ func main() {
...
+
+	// remove all stale lockfiles
+	for _, f := range lockFiles {
+		err := os.Remove(f)
+		if err != nil {
+			if errors.Is(err, os.ErrNotExist) {
+				continue
+			}
+
+			return fmt.Errorf("removing stale lock file %q", f)
```
Comment (mpimenov) [inline internal/cmd/git-combine/git-combine.go:483]:
```suggestion
			return fmt.Errorf("removing stale lock file %q: %w", f, err)
```

File: internal/cmd/git-combine/git-combine.go [hunk 2]
```diff
@@ -357,6 +360,11 @@ func doDaemon(dir string, done <-chan struct{}, opt Options) error {
 
 	opt.SetDefaults()
 
+	err := cleanupStaleLockFiles(dir)
+	if err != nil {
+		return fmt.Errorf("removing stale git lock files: %w", err)
```
Comment (keegancsmith) [inline internal/cmd/git-combine/git-combine.go:365]: if you are importing lib/errors, then use `errors.Wrap`. Same for other places in this PR I see you using fmt.Errorf. However, if this is just matching our existing style (which is possible) then nevermind :)

**Review by keegancsmith — APPROVED** [review]

File: internal/cmd/git-combine/git-combine.go [hunk 3]
```diff
@@ -440,6 +448,47 @@ func main() {
 	}
 }
 
+// cleanupStaleLockFiles removes any Git lock files that might have been left behind
+// by a crashed git-combine process
+func cleanupStaleLockFiles(gitDir string) error {
+	var lockFiles []string
+	for _, f := range []string{"git.pid.lock", "index.lock"} {
```
[the remark this answers is not in the document]
Comment (ggilmore) [reply internal/cmd/git-combine/git-combine.go]: You were correct, I should have written `gc.pid.lock. 

In addition, I have also added some more "lockfiles" from https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/cmd/gitserver/server/cleanup.go?L334, and added a comment in the code to document exactly where I found them.

File: internal/cmd/git-combine/git-combine.go (hunk 2 above)
Comment (ggilmore) [reply internal/cmd/git-combine/git-combine.go]: Done. I had forgotten about our internal errors package, and just used fmt.Errorf out of habit.

**Review by mpimenov — APPROVED** [review]

File: internal/cmd/git-combine/git-combine.go [hunk 4]
```diff
@@ -440,6 +448,63 @@ func main() {
...
+		if d.IsDir() || !strings.HasSuffix(path, ".lock") {
+			return nil
+		}
+
+		lockFiles = append(lockFiles, path)
+		return nil
+	})
+
+	if err != nil {
+		return errors.Wrapf(err, "finding stale lockfiles in %q: %w", refsDir)
```
Comment (mpimenov) [inline internal/cmd/git-combine/git-combine.go:488]: %w is redundant now

Comment (ggilmore) [reply internal/cmd/git-combine/git-combine.go:490]: whoops, yes - you're right

**Pull request merged**